File: wails-app/inventory-management/pkg/domain/product.go
package domain

import "time"

// Product represents a product in the inventory system
type Product struct {
	ProductID       int64      `json:"product_id"`
	Name            string     `json:"name"`
	Description     *string    `json:"description,omitempty"`
	SKU             string     `json:"sku"`
	CategoryID      int64      `json:"category_id"`
	BatchNo         *string    `json:"batch_no,omitempty"`
	ExpiryDate      *time.Time `json:"expiry_date,omitempty"`
	CostPrice       float64    `json:"cost_price"`
	SellingPrice    float64    `json:"selling_price"`
	CurrentQuantity int64      `json:"current_quantity"`
	MinStockLevel   int64      `json:"min_stock_level"`
	MaxStockLevel   *int64     `json:"max_stock_level,omitempty"`
	ReorderPoint    int64      `json:"reorder_point"`
	Unit            string     `json:"unit"` // pcs, kg, liter, box, carton
	ShelfLocation   *string    `json:"shelf_location,omitempty"`
	IsActive        bool       `json:"is_active"`
	CreatedAt       time.Time  `json:"created_at"`
	UpdatedAt       time.Time  `json:"updated_at"`
}

// ProductWithDetails extends Product with computed fields for API responses
type ProductWithDetails struct {
	Product
	CategoryName *string `json:"category_name,omitempty"`
}

File: wails-app/inventory-management-updated/frontend/src/types/product.ts
export interface Product {
    product_id: number;
    name: string;
    description?: string;
    sku: string;
    category_id: number;
    category_name?: string;
    batch_no?: string;
    expiry_date?: string;
    cost_price: number;
    selling_price: number;
    current_quantity: number;
    min_stock_level: number;
    max_stock_level?: number;
    reorder_point: number;
    unit: string;
    shelf_location?: string;
    is_active: boolean;
    created_at: string;
    updated_at?: string;
}

export interface ProductFormData {
    name: string;
    description?: string;
    sku: string;
    category_id: number | string;
    batch_no?: string;
    expiry_date?: string;
    cost_price: number | string;
    selling_price: number | string;
    current_quantity: number | string;
    min_stock_level: number | string;
    max_stock_level?: number | string;
    reorder_point: number | string;
    unit: string;
    shelf_location?: string;
}

export interface Category {
    category_id: number;
    name: string;
    description?: string;
    parent_category_id?: number;
    created_at: string;
}
